src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -286,6 +289,109 @@ private void AddStatementExceptBlockSuppressOperations(List<SuppressOperation> l
             AddSuppressWrappingIfOnSingleLineOperation(list, firstToken, lastToken);
         }
 
+        private void AddFormatSuppressOperations(List<SuppressOperation> list, SyntaxNode node)
+        {
+            foreach (var child in node.ChildNodesAndTokens())
```

Comment:
you could likely speed this up a lot by ignoring nodes that don't have `.ContainsDirectives` on them.